arduino/smartcar/smartcar.ino

```diff
@@ -101,3 +100,11 @@ void turnRight(){ // turns the car 10 degrees clockwise (degrees depend on TURNI
   turningAngle = turningAngle+TURNING_INCREMENT;
   car.setAngle(turningAngle);
 }
+
+void stopCar(){
```

Comment:
Why don't you measure the speed of the car instead via getSpeed and stop when that is 0?
But in any case, this loop blocking for 10 milliseconds for every iteration which is a huge period relatively speaking.
You should consider doing the same thing without blocking (ie no delays or blocking loops used).